### ISSUE TITLE
Bulk Response Null, Code 500

### ISSUE THREAD

**Issue by ltamrazov** [couchbaselabs/elasticsearch-transport-couchbase#72]
Hello, 

Getting this issue when running this version https://github.com/Branor/elasticsearch-transport-couchbase/releases/tag/v2.1.0-SNAPSHOT. Replication is not effected, and new documents get replicated successfully, but still get the error in ES and Couchbase:

elastic:
```
[2015-02-16 11:03:54,511][WARN ][org.eclipse.jetty.servlet.ServletHandler] /main_v1%2f21/_bulk_docs
java.lang.RuntimeException: indexing error, bulk response was null
	at org.elasticsearch.transport.couchbase.capi.ElasticSearchCAPIBehavior.bulkDocs(ElasticSearchCAPIBehavior.java:541)
	at com.couchbase.capi.servlet.CAPIServlet.handleBulkDocs(CAPIServlet.java:529)
	at com.couchbase.capi.servlet.CAPIServlet.service(CAPIServlet.java:89)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:847)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:594)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:485)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:119)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:486)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:233)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1065)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:412)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:192)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:999)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:117)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:111)
	at org.eclipse.jetty.server.Server.handle(Server.java:351)
	at org.eclipse.jetty.server.AbstractHttpConnection.handleRequest(AbstractHttpConnection.java:451)
	at org.eclipse.jetty.server.AbstractHttpConnection$RequestHandler.content(AbstractHttpConnection.java:931)
	at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:857)
	at org.eclipse.jetty.http.HttpParser.parseAvailable(HttpParser.java:235)
	at org.eclipse.jetty.server.AsyncHttpConnection.handle(AsyncHttpConnection.java:76)
	at org.eclipse.jetty.io.nio.SelectChannelEndPoint.handle(SelectChannelEndPoint.java:609)
	at org.eclipse.jetty.io.nio.SelectChannelEndPoint$1.run(SelectChannelEndPoint.java:45)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:599)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:534)
	at java.lang.Thread.run(Thread.java:745)
```

It does not seem to stop replication (adding new documents after we get this error works fine). Also it only happens with a bucket that has documents we excluded from replication using the following config:

```
# filter out all docs that don't match this
couchbase.keyFilter: org.elasticsearch.transport.couchbase.capi.RegexKeyFilter
couchbase.keyFilter.type: "include"
couchbase.keyFilter.keyFiltersRegex.idsToInclude: ^[updbrv]{1}-.+$
```

Not sure if its relevant, but the config also uses a typeSelector with multiple regexes. 

```couchbase.typeSelector: org.elasticsearch.transport.couchbase.capi.RegexTypeSelector```

Again, I'm pointing this out because we are replicating two buckets. In one, all the ids get replicated. The other contains ids that get replicated as well as ids that we filter out using the regex (that's the one that produces the error). 

couchdb.log prints:

```
[couchdb:error,2015-02-16T11:57:18.683,ns_1@couchbase.dev.mydomain.com:<0.31801.2>:couch_log:error:44]Replicator, request POST to "http://Administrator:*****@192.168.1.100:9091/main_v1%2f27/_bulk_docs" failed due to error {code,500}
```

debug.log: 

```
[error_logger:error,2015-02-16T11:57:50.716,ns_1@couchbase.dev.mydomain.com:error_logger<0.6.0>:ale_error_logger_handler:do_log:203]** Generic server <0.32498.2> terminating 
** Last message in was {'EXIT',<0.32511.2>,
                           {http_request_failed,"POST",
                               "http://Administrator:*****@192.168.1.100:9091/main_v1%2f41/_bulk_docs",
                               {error,{code,500}}}}
** When Server state == [{data,
                          [{"State",
                            {rep_state,
                             {rep,
                              <<"dc197740d72e32c7abc6b40b0890baf4/main/main_v1">>,
                              <<"main">>,
                              <<"/remoteClusters/dc197740d72e32c7abc6b40b0890baf4/buckets/main_v1">>,
                              "capi",
                              [{optimistic_replication_threshold,256},
                               {worker_processes,4},
                               {worker_batch_size,500},
                               {failure_restart_interval,30},
                               {doc_batch_size_kb,2048},
                               {checkpoint_interval,1800},
                               {max_concurrent_reps,16},
                               {connection_timeout,180},
                               {http_connections,20},
                               {retries_per_request,2},
                               {socket_options,
                                [{keepalive,true},{nodelay,false}]},
                               {pause_requested,false},
                               {supervisor_max_r,25},
                               {supervisor_max_t,5}]},
                             {rep_vb_status,41,<0.32498.2>,replicating},
                             {1424,105869,949679},
                             nil,nil,<0.28879.0>,<0.28876.0>,<<"main/41">>,
                             <<"*****@192.168.1.100:9091/main_v1%2f41">>,
                             {httpdb,
                              "*****@192.168.1.100:9091/main_v1%2f41/",nil,
                              [{"Accept","application/json"},
                               {"User-Agent","CouchDB/2.1.1r-432-gc2af28d"}],
                              180000,
                              [{connect_options,
                                [{keepalive,true},{nodelay,false}]},
                               {connect_timeout,180000},
                               {send_retry,0}],
                              2,250,xdc_lhttpc_pool,20},
                             {db,<0.459.0>,<0.460.0>,nil,
                              <<"1424098902710912">>,<0.456.0>,<0.472.0>,
                              {db_header,11,0,nil,nil,
                               <<0,0,0,0,161,242,0,0,0,0,19,46>>,
                               1,nil,nil},
                              0,
                              {btree,<0.456.0>,nil,
                               #Fun<couch_db_updater.7.78420415>,
                               #Fun<couch_db_updater.8.78420415>,
                               #Fun<couch_btree.1.39972947>,
                               #Fun<couch_db_updater.9.78420415>,1279,2558,
                               true},
                              {btree,<0.456.0>,nil,
                               #Fun<couch_db_updater.10.78420415>,
                               #Fun<couch_db_updater.11.78420415>,
                               #Fun<couch_db_updater.6.78420415>,
                               #Fun<couch_db_updater.12.78420415>,1279,2558,
                               true},
                              {btree,<0.456.0>,
                               {41458,<<>>,4910},
                               identity,identity,#Fun<couch_btree.1.39972947>,
                               nil,1279,2558,true},
                              0,<<"main/master">>,
                              "/Users/Levon/Library/Application Support/Couchbase/var/lib/couchdb/main/master.couch.2",
                              [],nil,
                              {user_ctx,null,[],undefined},
                              nil,
                              [before_header,after_header,on_file_open],
                              []},
                             <<"93eb051635834804844f1069ef444bd7">>,0,
                             undefined,0,0,22,22,0,245176386652781,
                             [22],
                             0,
                             {0,0},
                             "Mon, 16 Feb 2015 16:57:49 GMT",
                             {1424107669950445,#Ref<0.0.6.72623>},
                             {1424,105869,950200},
                             {1424,105869,949700},
                             [<0.32511.2>],
                             <0.32508.2>,false,
                             {"http://Administrator:kva%24im0d0@192.168.1.100:9091/",
                              [{<<"vb">>,41},
                               {<<"bucket">>,<<"main_v1">>},
                               {<<"bucketUUID">>,
                                <<"baa2747c512c4817bc52feecde9687ea">>}],
                              {httpdb,
                               "*****@192.168.1.100:9091/main_v1%2f41/",nil,
                               [{"Accept","application/json"},
                                {"User-Agent","CouchDB/2.1.1r-432-gc2af28d"}],
                               180000,
                               [{connect_options,
                                 [{keepalive,true},{nodelay,false}]},
                                {connect_timeout,180000},
                                {send_retry,0}],
                               2,250,xdc_lhttpc_pool,20}}}}]}]
** Reason for termination == 
** {http_request_failed,"POST",
                        "http://Administrator:*****@192.168.1.100:9091/main_v1%2f41/_bulk_docs",
                        {error,{code,500}}}

[error_logger:error,2015-02-16T11:57:50.718,ns_1@couchbase.dev.mydomain.com:error_logger<0.6.0>:ale_error_logger_handler:do_log:203]
=========================CRASH REPORT=========================
  crasher:
    initial call: xdc_vbucket_rep:init/1
    pid: <0.32498.2>
    registered_name: []
    exception exit: {http_request_failed,"POST",
                        "http://Administrator:*****@192.168.1.100:9091/main_v1%2f41/_bulk_docs",
                        {error,{code,500}}}
      in function  gen_server:terminate/6 (gen_server.erl, line 744)
    ancestors: [<0.28880.0>,<0.28876.0>,xdc_replication_sup,xdcr_sup,
                  ns_server_sup,ns_server_cluster_sup,<0.59.0>]
    messages: []
    links: [<0.28880.0>]
    dictionary: [{changes_reader,<0.32509.2>}]
    trap_exit: true
    status: running
    heap_size: 4185
    stack_size: 27
    reductions: 13740
  neighbours:

[error_logger:error,2015-02-16T11:57:50.835,ns_1@couchbase.dev.mydomain.com:error_logger<0.6.0>:ale_error_logger_handler:do_log:203]** Generic server <0.32520.2> terminating 
** Last message in was {'EXIT',<0.32530.2>,
                           {http_request_failed,"POST",
                               "http://Administrator:*****@192.168.1.100:9091/main_v1%2f32/_bulk_docs",
                               {error,{code,500}}}}
** When Server state == [{data,
                          [{"State",
                            {rep_state,
                             {rep,
                              <<"dc197740d72e32c7abc6b40b0890baf4/main/main_v1">>,
                              <<"main">>,
                              <<"/remoteClusters/dc197740d72e32c7abc6b40b0890baf4/buckets/main_v1">>,
                              "capi",
                              [{optimistic_replication_threshold,256},
                               {worker_processes,4},
                               {worker_batch_size,500},
                               {failure_restart_interval,30},
                               {doc_batch_size_kb,2048},
                               {checkpoint_interval,1800},
                               {max_concurrent_reps,16},
                               {connection_timeout,180},
                               {http_connections,20},
                               {retries_per_request,2},
                               {socket_options,
                                [{keepalive,true},{nodelay,false}]},
                               {pause_requested,false},
                               {supervisor_max_r,25},
                               {supervisor_max_t,5}]},
                             {rep_vb_status,32,<0.32520.2>,replicating},
                             {1424,105870,66191},
                             nil,nil,<0.28879.0>,<0.28876.0>,<<"main/32">>,
                             <<"*****@192.168.1.100:9091/main_v1%2f32">>,
                             {httpdb,
                              "*****@192.168.1.100:9091/main_v1%2f32/",nil,
                              [{"Accept","application/json"},
                               {"User-Agent","CouchDB/2.1.1r-432-gc2af28d"}],
                              180000,
                              [{connect_options,
                                [{keepalive,true},{nodelay,false}]},
                               {connect_timeout,180000},
                               {send_retry,0}],
                              2,250,xdc_lhttpc_pool,20},
                             {db,<0.459.0>,<0.460.0>,nil,
                              <<"1424098902710912">>,<0.456.0>,<0.472.0>,
                              {db_header,11,0,nil,nil,
                               <<0,0,0,0,161,242,0,0,0,0,19,46>>,
                               1,nil,nil},
                              0,
                              {btree,<0.456.0>,nil,
                               #Fun<couch_db_updater.7.78420415>,
                               #Fun<couch_db_updater.8.78420415>,
                               #Fun<couch_btree.1.39972947>,
                               #Fun<couch_db_updater.9.78420415>,1279,2558,
                               true},
                              {btree,<0.456.0>,nil,
                               #Fun<couch_db_updater.10.78420415>,
                               #Fun<couch_db_updater.11.78420415>,
                               #Fun<couch_db_updater.6.78420415>,
                               #Fun<couch_db_updater.12.78420415>,1279,2558,
                               true},
                              {btree,<0.456.0>,
                               {41458,<<>>,4910},
                               identity,identity,#Fun<couch_btree.1.39972947>,
                               nil,1279,2558,true},
                              0,<<"main/master">>,
                              "/Users/Levon/Library/Application Support/Couchbase/var/lib/couchdb/main/master.couch.2",
                              [],nil,
                              {user_ctx,null,[],undefined},
                              nil,
                              [before_header,after_header,on_file_open],
                              []},
                             <<"de3e8a21744649619e09154819e3a2e2">>,0,
                             undefined,0,0,23,23,0,256547314792210,
                             [2,23],
                             0,
                             {0,0},
                             "Mon, 16 Feb 2015 16:57:50 GMT",
                             {1424107670066990,#Ref<0.0.6.72762>},
                             {1424,105870,66713},
                             {1424,105870,66212},
                             [<0.32530.2>,<0.32531.2>],
                             <0.32527.2>,false,
                             {"http://Administrator:kva%24im0d0@192.168.1.100:9091/",
                              [{<<"vb">>,32},
                               {<<"bucket">>,<<"main_v1">>},
                               {<<"bucketUUID">>,
                                <<"baa2747c512c4817bc52feecde9687ea">>}],
                              {httpdb,
                               "*****@192.168.1.100:9091/main_v1%2f32/",nil,
                               [{"Accept","application/json"},
                                {"User-Agent","CouchDB/2.1.1r-432-gc2af28d"}],
                               180000,
                               [{connect_options,
                                 [{keepalive,true},{nodelay,false}]},
                                {connect_timeout,180000},
                                {send_retry,0}],
                               2,250,xdc_lhttpc_pool,20}}}}]}]
** Reason for termination == 
** {http_request_failed,"POST",
                        "http://Administrator:*****@192.168.1.100:9091/main_v1%2f32/_bulk_docs",
                        {error,{code,500}}}

[error_logger:error,2015-02-16T11:57:50.839,ns_1@couchbase.dev.mydomain.com:error_logger<0.6.0>:ale_error_logger_handler:do_log:203]
=========================CRASH REPORT=========================
  crasher:
    initial call: xdc_vbucket_rep:init/1
    pid: <0.32520.2>
    registered_name: []
    exception exit: {http_request_failed,"POST",
                        "http://Administrator:*****@192.168.1.100:9091/main_v1%2f32/_bulk_docs",
                        {error,{code,500}}}
      in function  gen_server:terminate/6 (gen_server.erl, line 744)
    ancestors: [<0.28880.0>,<0.28876.0>,xdc_replication_sup,xdcr_sup,
                  ns_server_sup,ns_server_cluster_sup,<0.59.0>]
    messages: [{'EXIT',<0.32531.2>,
                      {http_request_failed,"POST",
                          "http://Administrator:*****@192.168.1.100:9091/main_v1%2f32/_bulk_docs",
                          {error,{code,500}}}}]
    links: [<0.28880.0>]
    dictionary: [{changes_reader,<0.32528.2>}]
    trap_exit: true
    status: running
    heap_size: 4185
    stack_size: 27
    reductions: 13798
  neighbours:

[ns_server:debug,2015-02-16T11:58:12.295,ns_1@couchbase.dev.mydomain.com:compaction_new_daemon<0.448.0>:compaction_new_daemon:process_scheduler_message:1288]Starting compaction for the following buckets: 
[<<"main">>,<<"biz">>]
[ns_server:info,2015-02-16T11:58:12.295,ns_1@couchbase.dev.mydomain.com:<0.32665.2>:compaction_new_daemon:spawn_scheduled_kv_compactor:468]Start compaction of vbuckets for bucket main with config: 
[{database_fragmentation_threshold,{30,undefined}},
 {view_fragmentation_threshold,{30,undefined}}]
[ns_server:debug,2015-02-16T11:58:12.296,ns_1@couchbase.dev.mydomain.com:<0.32668.2>:compaction_new_daemon:bucket_needs_compaction:953]`main` data size is 23809, disk size is 2658412
[ns_server:info,2015-02-16T11:58:12.296,ns_1@couchbase.dev.mydomain.com:<0.32669.2>:compaction_new_daemon:spawn_scheduled_kv_compactor:468]Start compaction of vbuckets for bucket biz with config: 
[{database_fragmentation_threshold,{30,undefined}},
 {view_fragmentation_threshold,{30,undefined}}]
[ns_server:debug,2015-02-16T11:58:12.297,ns_1@couchbase.dev.mydomain.com:<0.32672.2>:compaction_new_daemon:bucket_needs_compaction:953]`biz` data size is 13768, disk size is 826375
[ns_server:debug,2015-02-16T11:58:12.297,ns_1@couchbase.dev.mydomain.com:compaction_new_daemon<0.448.0>:compaction_new_daemon:process_compactors_exit:1329]Finished compaction iteration.
[ns_server:debug,2015-02-16T11:58:12.297,ns_1@couchbase.dev.mydomain.com:compaction_new_daemon<0.448.0>:compaction_scheduler:schedule_next:60]Finished compaction too soon. Next run will be in 30s
[ns_server:debug,2015-02-16T11:58:13.241,ns_1@couchbase.dev.mydomain.com:compaction_new_daemon<0.448.0>:compaction_new_daemon:process_scheduler_message:1288]Starting compaction for the following buckets: 
[<<"main">>,<<"biz">>]
[ns_server:info,2015-02-16T11:58:13.242,ns_1@couchbase.dev.mydomain.com:<0.32705.2>:compaction_new_daemon:try_to_cleanup_indexes:564]Cleaning up indexes for bucket `main`
[ns_server:info,2015-02-16T11:58:13.243,ns_1@couchbase.dev.mydomain.com:<0.32705.2>:compaction_new_daemon:spawn_scheduled_views_compactor:494]Start compaction of indexes for bucket main with config: 
[{database_fragmentation_threshold,{30,undefined}},
 {view_fragmentation_threshold,{30,undefined}}]
[ns_server:info,2015-02-16T11:58:13.243,ns_1@couchbase.dev.mydomain.com:<0.32706.2>:compaction_new_daemon:try_to_cleanup_indexes:564]Cleaning up indexes for bucket `biz`
[ns_server:info,2015-02-16T11:58:13.244,ns_1@couchbase.dev.mydomain.com:<0.32706.2>:compaction_new_daemon:spawn_scheduled_views_compactor:494]Start compaction of indexes for bucket biz with config: 
[{database_fragmentation_threshold,{30,undefined}},
 {view_fragmentation_threshold,{30,undefined}}]
[ns_server:debug,2015-02-16T11:58:13.244,ns_1@couchbase.dev.mydomain.com:compaction_new_daemon<0.448.0>:compaction_new_daemon:process_compactors_exit:1329]Finished compaction iteration.
```

Thanks in advance, 

Levon

**Comment by ali2007** [comment]
HY 
do you remember me i asking you that the plugin dont exclude document from replication 
i find my error i m  raning couchbase 2.5 
i upgrade to 3.0
but now i have the same probleme like ltamrazov 
can you help me.
1.when i start replication i have its fan the pugin dont replicate the document with excludes ids
but es log this error 
java.lang.RuntimeException: indexing error, bulk response was null
2.plugin replicate 1025 docs(vbuckets) by default how can i stop this 
thinks branor

**Comment by ltamrazov** [comment]
@Branor Hi, any update on when this release will be out?

Thanks!

**Comment by ali2007** [comment]
I think that the errors comming because i use a snapshot i dont know how to
have dependencie with maven

**Comment by ali2007** [comment]
For vbuckets document i use the filter exclude to type couchbasechekpoints

**Comment by Branor** [comment]
Fixed in v2.1.1, check out the RC here: https://github.com/couchbaselabs/elasticsearch-transport-couchbase/releases/tag/2.1.1-RC
